google-cloud-storage/src/test/java/com/google/cloud/storage/GrpcTransformPageDecoratorTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.core.ApiFuture;
+import com.google.api.core.ApiFutures;
+import com.google.api.gax.paging.AbstractPage;
+import com.google.api.gax.rpc.ApiCallContext;
+import com.google.api.gax.rpc.PageContext;
+import com.google.api.gax.rpc.PagedListDescriptor;
+import com.google.api.gax.rpc.UnaryCallable;
+import com.google.cloud.storage.GrpcStorageImpl.TransformingPageDecorator;
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class GrpcTransformPageDecoratorTest {
+
+  @Test
+  public void valueTranslationTest() {
+    // Initial values for the first page
+    List<String> initialValues = Arrays.asList("string1", "string2", "string3");
+    // Request which will be appended to the second page
+    String request = "string4";
+
+    // Expected values after the translation
+    List<String> expectedValuesPageOne = Arrays.asList("STRING1", "STRING2", "STRING3");
+    List<String> expectedValuesPageTwo = Arrays.asList("STRING4");
+    List<String> expectedPagesValuesMerged =
+        Stream.of(expectedValuesPageOne, expectedValuesPageTwo)
+            .flatMap(list -> list.stream())
+            .collect(Collectors.toList());
+
+    StringPagedListDescriptor descriptor = new StringPagedListDescriptor();
+    UnaryCallable callable = new StringPagedCallable();
+    PageContext<String, List<String>, String> context =
+        PageContext.create(callable, descriptor, request, Mockito.mock(ApiCallContext.class));
+    ListStringPage page = new ListStringPage(context, initialValues);
+    TransformingPageDecorator decorator =
```

Comment:
Missing type parameters
```suggestion
    TransformingPageDecorator<String, List<String>, String, ListStringPage, String> decorator =
```